tests/integration_tests/document_loaders/test_mongodb.py

```diff
@@ -0,0 +1,121 @@
+        {"_id": "2", "address": {"building": "2", "room": "2"}},
+    ]
+
+
+@pytest.fixture
+def expected_documents():
+    return [
+        Document(
+            page_content="{'_id': '1', 'address': {'building': '1', 'room': '1'}}",
+            metadata={"database": "sample_restaurants", "collection": "restaurants"},
+        ),
+        Document(
+            page_content="{'_id': '2', 'address': {'building': '2', 'room': '2'}}",
+            metadata={"database": "sample_restaurants", "collection": "restaurants"},
+        ),
+    ]
+
+
+@pytest.mark.asyncio
+async def test_async_load_mocked(mocker, docs, expected_documents):
+    async def mock_find():
+        for doc in docs:
+            yield doc
+
+    mock_collection = MagicMock()
+    mock_db = MagicMock()
+    mock_client = MagicMock()
+
+    mocker.patch.object(mock_collection, "find", return_value=mock_find())
+    mocker.patch.object(AsyncIOMotorClient, "get_database", return_value=mock_db)
+    mocker.patch.object(mock_db, "get_collection", return_value=mock_collection)
+
+    with patch("motor.motor_asyncio.AsyncIOMotorClient", return_value=mock_client):
+        loader = MongodbLoader(
+            "mongodb://localhost:27017", "sample_restaurants", "restaurants"
+        )
+
+    documents = await loader._async_load()
+
+    assert documents == expected_documents
+
+
+@pytest.mark.asyncio
+async def test_async_partial_load_mocked(mocker, docs, expected_documents):
+    async def mock_find():
+        for doc in docs:
+            yield doc
+
+    expected_documents.remove(expected_documents[1])
+
+    mock_collection = MagicMock()
+    mock_db = MagicMock()
+    mock_client = MagicMock()
+
+    mocker.patch.object(mock_collection, "find", return_value=mock_find())
+    mocker.patch.object(AsyncIOMotorClient, "get_database", return_value=mock_db)
+    mocker.patch.object(mock_db, "get_collection", return_value=mock_collection)
+
+    with patch("motor.motor_asyncio.AsyncIOMotorClient", return_value=mock_client):
+        loader = MongodbLoader(
+            "mongodb://localhost:27017", "sample_restaurants", "restaurants"
+        )
+
+    with pytest.raises(Exception) as error_PartialLoad:
+        await loader._async_load()
+
+        assert (
+            str(error_PartialLoad.value)
+            == "Error: Only partial collection of documents returned."
+        )
+
+
+def test_load_mocked(expected_documents):
+    mock_async_load = AsyncMock()
+
+    mock_async_load.return_value = expected_documents
+
+    with patch("langchain.document_loaders.MongodbLoader", MagicMock()):
+        loader = MongodbLoader(
+            "mongodb://localhost:27017", "test_db", "test_collection"
+        )
+
+    loader._async_load = mock_async_load
+
+    documents = loader.load()
+
+    assert documents == expected_documents
+
+
+@pytest.mark.skipif(not mongo_api_key_set, reason="MONGODB_API_KEY not provided.")
+def test_load_actual_db() -> None:
```

Comment:
Not sure if this is a good idea, but maybe worth to share. I know we have docker enabled, maybe we can set up mongodb in docker which can make real testing much easier and cheaper. (We don’t need to include it here as it will make this PR much more complex. Generally we want to have each PR serve a single purpose. But it could be a good idea to explore in a following PR. :))

Reply:
@skcoirz including less to be included in the integration_test pipeline, more just so people can use as a harness in their own mongo implementation.  I used this test case to verify I was pulling in the 25k restaurants from the sample_restaurants mongo instance.  This will skip on the CI pipeline.  I don't mind pulling out of the script, but I felt it might be useful for some.

Reply:
I see. From what I can see, if it’s doable in CI, langchain usually test it in CI with related setup; otherwise it covers the test as an example in jupyter notebook. There are usually sample files for multi-tooling usage, such as the titanic.csv file in agent examples.

Reply:
got it.  what do you suggest for this test case?  the database i use is the sample db for Mongo (I've tested with other databases as well).  if there's a desire to spin up a docker container to verify on real data, that works.  curious if you want to do that as part of this PR?

Reply:
Is there anything we can re-use in the existing example docs? 
https://github.com/search?q=repo%3Ahwchase17%2Flangchain%20docs%2Fmodules%2Findexes%2Fdocument_loaders%2Fexamples%2Fexample_data&type=code

Reply:
Not really.  I see CSVs, JSON, etc.  but the mongo driver hooks into the mongodb and retrieves docs.  if the CI wanted to spin up a mongodb docker container, then seed the db, that's an option.  I just didn't want to go down that route as it seemed inconsistent with other Document Loader designs